ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -203,6 +208,35 @@ tests =
 traceFromList :: [a] -> Trace (SimResult ()) a
 traceFromList = Trace.fromList (MainReturn  (Time 0) (Labelled (ThreadId []) (Just "main")) () [])
 
+testWithIOSim :: (SimTrace Void -> Int -> Property)
+              -> Int
+              -> AbsBearerInfo
+              -> DiffusionScript
+              -> Property
+testWithIOSim f traceNumber bi ds =
+  let sim :: forall s . IOSim s Void
+      sim = diffusionSimulation (toBearerInfo bi)
+                                ds
+                                iosimTracer
+   in labelDiffusionScript ds
+    $ f (runSimTrace sim) traceNumber
```

Comment:
You could simplify the type signature to:
```
testWithIOSim :: (SimTrace Void -> Property)
              -> AbsBearerInfo
              -> DiffusionScript
              -> Property
```
and have the `traceNumber` defined: either inside `f` or passed to `f` if different values are used whether we run it with `IOSim` or `IOSimPOR`.

The same with `testWithIOSimPOR`.

Reply:
I know, I made the tradeoff of being explicit about the `traceNumber` when defining the test because some tests do require a different number of tests and it is easier to tweak in the future if we want. They are also available at the top level instead of having to go to each function manually change the value inside `f`

Reply:
It's still manual, just in a different place.

Reply:
Yes but the place it is at now is more upfront in the sense that they are all in the same place: the test tree.